pkg/registry/generic/etcd/etcd.go

```diff
@@ -295,22 +307,31 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 		}
 
 		creating = false
-		newVersion, err := e.Helper.Versioner.ObjectResourceVersion(obj)
-		if err != nil {
-			return nil, nil, err
-		}
-		if newVersion != version {
-			// TODO: return the most recent version to a client?
-			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("the object has been modified; please apply your changes to the latest version and try again"))
+		if useLatestResourceVersion {
+			// Update the object's resource version to match the latest etcd object's resource version.
+			err = e.Helper.Versioner.UpdateObject(obj, node)
```

Comment:
I think we can nuke update object if we fix events - there were only a few callers last I checked.

> On Jun 19, 2015, at 3:55 PM, Daniel Smith <notifications@github.com> wrote:
> 
> In pkg/registry/generic/etcd/etcd.go:
> 
> > @@ -295,22 +307,31 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
> >  		}
> >  
> >  		creating = false
> > -		newVersion, err := e.Helper.Versioner.ObjectResourceVersion(obj)
> > -		if err != nil {
> > -			return nil, nil, err
> > -		}
> > -		if newVersion != version {
> > -			// TODO: return the most recent version to a client?
> > -			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("the object has been modified; please apply your changes to the latest version and try again"))
> > +		if useLatestResourceVersion {
> > +			// Update the object's resource version to match the latest etcd object's resource version.
> > +			err = e.Helper.Versioner.UpdateObject(obj, node)
> You already have the latest version; so use 'version', not node-- if that means you need to fix/duplicate UpdateObject, then so be it. :)
> 
> —
> Reply to this email directly or view it on GitHub.
>